internal/services/labservice/lab_service_lab_plan_resource.go

```diff
@@ -0,0 +1,659 @@
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/labservices/2022-08-01/labplan"
+	azValidate "github.com/hashicorp/terraform-provider-azurerm/helpers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	computeValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/labservice/validate"
+	networkValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type LabServiceLabPlanModel struct {
+	Name                   string                `tfschema:"name"`
+	ResourceGroupName      string                `tfschema:"resource_group_name"`
+	Location               string                `tfschema:"location"`
+	AllowedRegions         []string              `tfschema:"allowed_regions"`
+	DefaultAutoShutdown    []DefaultAutoShutdown `tfschema:"default_auto_shutdown"`
+	DefaultConnection      []DefaultConnection   `tfschema:"default_connection"`
+	DefaultNetworkSubnetId string                `tfschema:"default_network_subnet_id"`
+	SharedGalleryId        string                `tfschema:"shared_gallery_id"`
+	Support                []Support             `tfschema:"support"`
+	Tags                   map[string]string     `tfschema:"tags"`
+}
+
+type DefaultAutoShutdown struct {
+	DisconnectDelay                 string                     `tfschema:"disconnect_delay"`
+	IdleDelay                       string                     `tfschema:"idle_delay"`
+	NoConnectDelay                  string                     `tfschema:"no_connect_delay"`
+	ShutdownOnDisconnectEnabled     bool                       `tfschema:"shutdown_on_disconnect_enabled"`
+	ShutdownOnIdle                  labplan.ShutdownOnIdleMode `tfschema:"shutdown_on_idle"`
+	ShutdownWhenNotConnectedEnabled bool                       `tfschema:"shutdown_when_not_connected_enabled"`
+}
+
+type DefaultConnection struct {
+	ClientRdpAccess labplan.ConnectionType `tfschema:"client_rdp_access"`
+	ClientSshAccess labplan.ConnectionType `tfschema:"client_ssh_access"`
+	WebRdpAccess    labplan.ConnectionType `tfschema:"web_rdp_access"`
+	WebSshAccess    labplan.ConnectionType `tfschema:"web_ssh_access"`
+}
+
+type Support struct {
+	Email        string `tfschema:"email"`
+	Instructions string `tfschema:"instructions"`
+	Phone        string `tfschema:"phone"`
+	Url          string `tfschema:"url"`
+}
+
+type LabServiceLabPlanResource struct{}
+
+var _ sdk.ResourceWithUpdate = LabServiceLabPlanResource{}
+
+func (r LabServiceLabPlanResource) ResourceType() string {
+	return "azurerm_lab_service_lab_plan"
```

Comment:
This should be called 
```suggestion
	return "azurerm_lab_service_plan"
```

Reply:
Updated